rollup/internal/controller/watcher/chunk_proposer_test.go

```diff
@@ -141,6 +145,18 @@ func testChunkProposerLimits(t *testing.T) {
 			expectedChunksLen:          1,
 			expectedBlocksInFirstChunk: 1,
 		},
+		{
+			name:                       "ForkBlockReached",
+			maxBlockNum:                100,
+			maxTxNum:                   10000,
+			maxL1CommitGas:             50000000000,
+			maxL1CommitCalldataSize:    1000000,
+			maxRowConsumption:          1000000,
+			chunkTimeoutSec:            1000000000000,
+			expectedChunksLen:          1,
```

Comment:
I think this unit test don't cover the hard fork logic.

Reply:
Can you elaborate? I am fairly certain they cover the changes. These tests fail without the changes I made in chunk and batch proposers.

Reply:
Yes, you are right.

chunk: 
<img width="921" alt="image" src="https://github.com/scroll-tech/scroll/assets/6947653/d12dadda-eeea-42fd-ac47-090353866311">

batch:
<img width="812" alt="image" src="https://github.com/scroll-tech/scroll/assets/6947653/f67013a7-1f85-4830-a0cb-dbaecab85a3e">